### PR TITLE
Add priority class to the daemonsets

### PR DESCRIPTION
**Pull request by Michael-Sinz**
Add a priority class for omsagent and have the daemonsets use this
to be sure to schedule the pods.

Daemonset pods are constrained in scheduling to run on specific
nodes.  This is done by the daemonset controller.  When a node shows
up it will create a pod with a strong affinity to that node.  When a
node goes away, it will delete the pod with the node affinity to that
node.

Kubernetes pod scheduling does not know it is a daemonset but it does
know it is tied to a specific node.  With default scheduling, it is
possible for the pods to be "frozen out" of a node because the node
already is full.  This can happen because "normal" pods may already
exist and are looking for a node to get scheduled on when a node is
added to the cluster.  The daemonset controller will only first
create the pod for the node at around the same time.  The kubernetes
scheduler is running async from all of this and thus there can be a
race as to who gets scheduled on the node.

The pod priority class (and thus the pod priority) is a way to indicate
that the pod has a higher scheduling priority than a default pod.

By default, all pods are at priority 0.  Higher numbers are higher
priority.  Setting the priority to something greater than zero will
allow the omsagent daemonsets to win a race against "normal" pods for
scheduled resources on a node - and will also allow for graceful
eviction in the case the node is too full.

Without this, omsagent can be left out of node in clusters that are
very busy, especially in dynamic scaling situations.

I did not test the windows pod as we have no windows clusters.

Fixes issue #391

### CODE REVIEW

**Comment by Michael-Sinz** [comment]
I have actually not tested this in a significant way yet due to our deployment depending on having the chart in a helm server somewhere.  I am still working on an adjustment to the deployment process to allow me to reference my local changes.

**Review by mradmila — APPROVED** [review]
You should probably not include formatting changes and you should link the 391 issue.

**Review by ganga1980 — DISMISSED** [review]
lgtm. After the PR merged, automatically deploys to our dev clusters (which has windows nodes ) so I can help on validating on windows nodes.

File: charts/azuremonitor-containers/values.yaml
Comment (vishiy) [inline charts/azuremonitor-containers/values.yaml:35]: Why is default 10 ? Can we have 0 (no priority) if not specified thru params ?

File: charts/azuremonitor-containers/templates/omsagent-priorityclass.yaml
Comment (vishiy) [inline charts/azuremonitor-containers/templates/omsagent-priorityclass.yaml:19]: check needed for ensuring .Values.omsagent.priority is a number ?

Comment (vishiy) [inline charts/azuremonitor-containers/templates/omsagent-priorityclass.yaml:19]: Can you also please add this param to the .README.MD in the repo (relative path - charts/azuremonitor-containers/README.md)

File: charts/azuremonitor-containers/values.yaml
Comment (Michael-Sinz) [reply charts/azuremonitor-containers/values.yaml:35]: The reason for something larger than 0 is that at 0 this will end up having pods for some nodes stay pending due to the race I discussed if the cluster is dynamically scaling due to pending work pods.  If k8s had thought of this from the start, daemonset controller would have defaulted to something like 10 to begin with such that daemonsets are not frozen out.

File: charts/azuremonitor-containers/templates/omsagent-priorityclass.yaml
Comment (Michael-Sinz) [reply charts/azuremonitor-containers/templates/omsagent-priorityclass.yaml:19]: Will add something to the readme.  Good idea.

Comment (Michael-Sinz) [reply charts/azuremonitor-containers/templates/omsagent-priorityclass.yaml:19]: Helm will do those checks as part of the validation against the schema when applying to a cluster.

**Comment by Michael-Sinz** [comment]
@ganga1980 - needs another review as I updated the README.md to describe the priority value.

**Review by ganga1980 — APPROVED** [review]

**Review by vishiy — APPROVED** [review]

**Pull request merged**
